{"kind":"symbol","metadata":{"roleHeading":"Class","modules":[{"name":"SnowplowTracker"}],"fragments":[{"text":"class","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"MediaErrorEvent"}],"externalID":"c:@M@SnowplowTracker@objc(cs)SPMediaErrorEvent","symbolKind":"class","title":"MediaErrorEvent","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"MediaErrorEvent"}]},"sections":[],"relationshipsSections":[{"identifiers":["doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/SelfDescribingAbstract"],"title":"Inherits From","kind":"relationships","type":"inheritsFrom"},{"type":"conformsTo","title":"Conforms To","kind":"relationships","identifiers":["doc:\/\/SnowplowTracker\/objc(pl)NSObject","doc:\/\/SnowplowTracker\/s7CVarArgP","doc:\/\/SnowplowTracker\/s28CustomDebugStringConvertibleP","doc:\/\/SnowplowTracker\/s23CustomStringConvertibleP","doc:\/\/SnowplowTracker\/SQ","doc:\/\/SnowplowTracker\/SH"]}],"hierarchy":{"paths":[["doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker"]]},"schemaVersion":{"major":0,"patch":0,"minor":3},"identifier":{"url":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Media player event tracked when the resource could not be loaded due to an error."}],"primaryContentSections":[{"declarations":[{"platforms":["macOS"],"tokens":[{"kind":"attribute","text":"@objc"},{"kind":"text","text":"(SPMediaErrorEvent) "},{"kind":"keyword","text":"class"},{"text":" ","kind":"text"},{"kind":"identifier","text":"MediaErrorEvent"}],"languages":["swift"]}],"kind":"declarations"}],"topicSections":[{"identifiers":["doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/init(errorCode:errorName:errorDescription:)"],"title":"Initializers"},{"identifiers":["doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorCode","doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorDescription","doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorName"],"title":"Instance Properties"},{"title":"Instance Methods","identifiers":["doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorCode(_:)","doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorDescription(_:)","doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorName(_:)"]}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/snowplowtracker\/mediaerrorevent"]}],"references":{"doc://SnowplowTracker/s28CustomDebugStringConvertibleP":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/s28CustomDebugStringConvertibleP","title":"Swift.CustomDebugStringConvertible"},"doc://SnowplowTracker/s7CVarArgP":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/s7CVarArgP","title":"Swift.CVarArg"},"doc://SnowplowTracker/SQ":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/SQ","title":"Swift.Equatable"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorCode":{"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorCode","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"errorCode"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"kind":"symbol","url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errorcode","abstract":[{"type":"text","text":"Error-identifying code for the playback issue. E.g. E522"}],"role":"symbol","title":"errorCode","type":"topic"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent":{"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent","title":"MediaErrorEvent","fragments":[{"text":"class","kind":"keyword"},{"kind":"text","text":" "},{"text":"MediaErrorEvent","kind":"identifier"}],"navigatorTitle":[{"text":"MediaErrorEvent","kind":"identifier"}],"abstract":[{"text":"Media player event tracked when the resource could not be loaded due to an error.","type":"text"}],"role":"symbol","type":"topic","url":"\/documentation\/snowplowtracker\/mediaerrorevent","kind":"symbol"},"doc://SnowplowTracker/documentation/SnowplowTracker":{"title":"SnowplowTracker","role":"collection","kind":"symbol","abstract":[],"url":"\/documentation\/snowplowtracker","identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker","type":"topic"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorCode(_:)":{"title":"errorCode(_:)","abstract":[{"type":"text","text":"Error-identifying code for the playback issue. E.g. E522"}],"url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errorcode(_:)","kind":"symbol","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"errorCode"},{"text":"(","kind":"text"},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"text":"?) -> ","kind":"text"},{"text":"Self","kind":"typeIdentifier"}],"type":"topic","role":"symbol","identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorCode(_:)"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/init(errorCode:errorName:errorDescription:)":{"title":"init(errorCode:errorName:errorDescription:)","role":"symbol","kind":"symbol","type":"topic","abstract":[],"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/init(errorCode:errorName:errorDescription:)","fragments":[{"text":"init","kind":"identifier"},{"kind":"text","text":"("},{"kind":"externalParam","text":"errorCode"},{"text":": ","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"},{"kind":"text","text":"?, "},{"text":"errorName","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"},{"kind":"text","text":"?, "},{"kind":"externalParam","text":"errorDescription"},{"text":": ","kind":"text"},{"text":"String","preciseIdentifier":"s:SS","kind":"typeIdentifier"},{"text":"?)","kind":"text"}],"url":"\/documentation\/snowplowtracker\/mediaerrorevent\/init(errorcode:errorname:errordescription:)"},"doc://SnowplowTracker/s23CustomStringConvertibleP":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/s23CustomStringConvertibleP","title":"Swift.CustomStringConvertible"},"doc://SnowplowTracker/SH":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/SH","title":"Swift.Hashable"},"doc://SnowplowTracker/documentation/SnowplowTracker/SelfDescribingAbstract":{"title":"SelfDescribingAbstract","fragments":[{"text":"class","kind":"keyword"},{"kind":"text","text":" "},{"text":"SelfDescribingAbstract","kind":"identifier"}],"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/SelfDescribingAbstract","role":"symbol","type":"topic","kind":"symbol","navigatorTitle":[{"text":"SelfDescribingAbstract","kind":"identifier"}],"abstract":[{"type":"text","text":"The properties for all the self-describing events."}],"url":"\/documentation\/snowplowtracker\/selfdescribingabstract"},"doc://SnowplowTracker/objc(pl)NSObject":{"type":"unresolvable","identifier":"doc:\/\/SnowplowTracker\/objc(pl)NSObject","title":"ObjectiveC.NSObjectProtocol"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorName(_:)":{"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorName(_:)","title":"errorName(_:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"text":"errorName","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":"?) -> ","kind":"text"},{"text":"Self","kind":"typeIdentifier"}],"abstract":[{"type":"text","text":"Name for the type of error that occurred in the playback. E.g. forbidden"}],"role":"symbol","type":"topic","url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errorname(_:)","kind":"symbol"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorDescription(_:)":{"abstract":[{"text":"Longer description for the error that occurred in the playback.","type":"text"}],"kind":"symbol","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorDescription"},{"kind":"text","text":"("},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"text":"?) -> ","kind":"text"},{"kind":"typeIdentifier","text":"Self"}],"role":"symbol","url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errordescription(_:)","identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorDescription(_:)","type":"topic","title":"errorDescription(_:)"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorDescription":{"type":"topic","title":"errorDescription","url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errordescription","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorDescription"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"},{"text":"?","kind":"text"}],"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorDescription","kind":"symbol","abstract":[{"type":"text","text":"Longer description for the error that occurred in the playback."}],"role":"symbol"},"doc://SnowplowTracker/documentation/SnowplowTracker/MediaErrorEvent/errorName":{"identifier":"doc:\/\/SnowplowTracker\/documentation\/SnowplowTracker\/MediaErrorEvent\/errorName","title":"errorName","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"errorName","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:SS","kind":"typeIdentifier","text":"String"},{"text":"?","kind":"text"}],"abstract":[{"text":"Name for the type of error that occurred in the playback. E.g. forbidden","type":"text"}],"role":"symbol","type":"topic","url":"\/documentation\/snowplowtracker\/mediaerrorevent\/errorname","kind":"symbol"}}}